tests/x11/rrdtool_x11.pm

```diff
@@ -0,0 +1,87 @@
+# SUSE"s openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Regression test for rrdtool
+# - created rrd file
+# - update rrd file
+# - fetch data from file
+# - generate images with different options
+# - compare images with needles
+#
+# Expected result:
+# 4 graphics will be displayed without error.
+#
+# Maintainer: Marcelo Martins <mmartins@suse.cz>
+
+use base "x11test";
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+my $imagespeed = "";
+
+# open images and check needles.
+sub open_image {
+    send_key("alt-f2");
+    type_string("eog /tmp/rrdtool/$imagespeed.png\n");
+    sleep 5;
+    assert_screen "rrdtool_image";
+    send_key 'alt-f4';
+    assert_screen "rrdtool_xterm";
+    sleep 5;
+}
+
+sub run {
+    select_console 'x11';
+    x11_start_program('xterm');
+    become_root;
+    pkcon_quit;
+    # create a tmp dir/files to work
+    assert_script_run "mkdir /tmp/rrdtool; cd /tmp/rrdtool";
+    # install requirements
+    zypper_call "in rrdtool";
+    # create a rrd file
+    assert_script_run "rrdtool create test.rrd --start 920804400  DS:speed:COUNTER:600:U:U RRA:AVERAGE:0.5:1:24 RRA:AVERAGE:0.5:6:10";
+    # update the rrd file
+    assert_script_run "rrdtool update test.rrd 920804700:12345 920805000:12357 920805300:12363 ; /
+rrdtool update test.rrd 920805600:12363 920805900:12363 920806200:12373 ; /
+rrdtool update test.rrd 920806500:12383 920806800:12393 920807100:12399 ; /
+rrdtool update test.rrd 920807400:12405 920807700:12411 920808000:12415 ; /
+rrdtool update test.rrd 920808300:12420 920808600:12422 920808900:12423";
+    # fetch the rrd file
+    assert_script_run "rrdtool fetch test.rrd AVERAGE --start 920804400 --end 920809200";
+    # make a graph 1
+    assert_script_run "rrdtool graph speed.png --start 920804400 --end 920808000 DEF:myspeed=test.rrd:speed:AVERAGE LINE2:myspeed#FF0000";
+    #Open image
+    $imagespeed = "speed";
+    open_image;
+    # make a graph 2
+    assert_script_run "rrdtool graph speed2.png --start 920804400 --end 920808000 --vertical-label m/s DEF:myspeed=test.rrd:speed:AVERAGE CDEF:realspeed=myspeed,1000,* LINE2:realspeed#FF0000";
+    #Open image
+    $imagespeed = "speed2";
+    open_image;
+    # make a graph 3
+    # On SLES15 need this extra sleep to next comando run correct.
+    sleep 15;
```

Comment:
open_image assert screen rrdtool_xterm, why do you need to wait 15 seconds after it? If you need to wait for something longer time, it is better to do it in loop with smaller sleeps and check for some condition to end it. See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/hpc/ganglia_client.pm#L41

Reply:
If this workaround is only for SLE15, consider using `is_sle('=15')` There is already sleep in the `open_image`, does that mean you need 5 seconds for sle12 and 20 for sle15?

Reply:
yes, I added is_sle('15+'), and tested with 15s, working fine.